include/no_os_pwm.h

```diff
@@ -99,6 +99,8 @@ struct no_os_pwm_init_param {
 struct no_os_pwm_desc {
 	/** PWM mutex*/
 	void *mutex;
+	/** Hardware version necessary for checking for the right register offsets  */
+	uint32_t hw_major_ver;
```

Comment:
So i'm wondering about adding this here, it seems unfit. This is a top level API header and such a change has implications in all pwm implementations.

Either rename it to something very generically named like `version` which would apply to other implementations as well, or move this field to the axi_pwm implementation of this API, here: https://github.com/analogdevicesinc/no-OS/blob/main/drivers/axi_core/axi_pwmgen/axi_pwm_extra.h#L69